### PR TITLE
Added geometric function to determine distance

### PR DESCRIPTION
**Pull request by theis999**

### CODE REVIEW

File: control_info.h [hunk 1]
```diff
@@ -241,6 +241,13 @@ local function item_is_tile(item)
     return false
 end
 
+local function tile_is_in_reach()	
+	local x = player.position.x - target_position[1]
+	local y = player.position.y - target_position[2]
+	local dis = math.sqrt(x^2+y^2) --sqrt(a^2+b)^=sqrt(c^2)
```
Comment (MortenTobiasNielsen) [inline control_info.h:247]: --sqrt(a^2+b^2)=sqrt(c^2)

File: control_info.h [hunk 2]
```diff
@@ -241,6 +241,13 @@ local function item_is_tile(item)
...
+	return dis < 11 -- reach is 10 tiles
```
Comment (MortenTobiasNielsen) [inline control_info.h:248]: I think Return dis <= 10 would be more accurate

File: control_info.h [hunk 3]
```diff
@@ -257,13 +264,17 @@ local function build()
 
 	if (item ~= "rail") then
 		if item_is_tile(item) then
+			if not tile_is_in_reach() then
```
Comment (MortenTobiasNielsen) [inline control_info.h:267]: We need to check if the player is currently walking and first give this warning, when that is not the case. Like it is done for placing buildings.

**Review by MortenTobiasNielsen — CHANGES_REQUESTED** [review]

File: control_info.h (hunk 1 above)
Comment (theis999) [reply control_info.h:247]: Yeah, my bad.

File: control_info.h (hunk 2 above)
Comment (theis999) [reply control_info.h:248]: Fair, i don't have a preference either way.

File: control_info.h (hunk 3 above)
Comment (theis999) [reply control_info.h:267]: You are right

File: control_info.h (hunk 2 above)
Comment (MortenTobiasNielsen) [reply control_info.h:248]: It is not so much about preference as it is about what is correct game mechanics - I want it to be true to the game. 😊

**Pull request merged**